Sage/carbon
# feat(progress-tracker): describe progress-tracker using design tokens, remove variant prop (FE-4455)

BREAKING CHANGE: Variant prop has been deprecated from the component.

### Proposed behaviour

Describes the progress-tracker component using design tokens.
Removes a `variant` prop from a component.
Removes `variant - default` and `variant-traffic` from the storybook documentation and adds a new one `color variants` showing the color change according to progress.
Adds missing props in the progress-tracker.style.js file.
Updates tests that are related to the changes made.

### Current behaviour

The progress-tracker component uses the theme styled-component property.
The propTypes in the css files are missing the props used.
The component has a variant prop with two options: `default` and `traffic`.

### Checklist




#### QA


### Additional context
https://confluence.sage.com/display/DSY/Progress+tracker

![image](https://user-images.githubusercontent.com/44143630/149545917-d00f4ddd-87fe-4488-901d-e7fb0b7cdbca.png)

### Testing instructions

The correct appearance can be checked in the component: Progress Tracker.
For theme: `mint` and `aegean` there should be no differences.
Use devtools to check if progress tracker use css variables.
Removed variant: `default` and `traffic` from storybook documentation. Added color variants.
